### PR TITLE
Add flutter startup benchmark

### PR DESCRIPTION
**Pull request by jensjoha**
Attempt of adding a flutter startup benchmark as directed on
https://github.com/flutter/flutter/pull/111461#issuecomment-1247113905

I don't know if there's any way of verifying that it works, though?

### CODE REVIEW

File: TESTOWNERS [hunk 1]
```diff
@@ -219,6 +219,9 @@
 /dev/devicelab/bin/tasks/flutter_gallery_macos__start_up.dart @a-wallen @flutter/desktop
 /dev/devicelab/bin/tasks/flutter_gallery_win_desktop__compile.dart @yaakovschectman @flutter/desktop
 /dev/devicelab/bin/tasks/flutter_gallery_win_desktop__start_up.dart @yaakovschectman @flutter/desktop
+/dev/devicelab/bin/tasks/flutter_tool_startup__windows.dart @jensj @flutter/tool
+/dev/devicelab/bin/tasks/flutter_tool_startup__linux.dart @jensj @flutter/tool
+/dev/devicelab/bin/tasks/flutter_tool_startup__macos.dart @jensj @flutter/tool
```
Comment (keyonghan) [inline TESTOWNERS:224]: These owner are expected to be github username.

**Review by keyonghan — COMMENTED** [review]
> I don't know if there's any way of verifying that it works, though?

You can validate in different ways:
1) locally: https://github.com/flutter/flutter/tree/master/dev/devicelab#running-tests-locally
2) via `led` run: go/flutter-luci-recipes#testing-a-local-change-for-a-new-builderrecipe

**Review by jonahwilliams — APPROVED** [review]
LGTM to me, but wait for @keyonghan too.

If they happen to fail on CI, they're marked bringup so they won't close the tree and we can investigate afterwards

File: TESTOWNERS (hunk 1 above)
Comment (jensjoha) [reply TESTOWNERS]: Done.

File: .ci.yaml [hunk 2]
```diff
@@ -4441,3 +4441,33 @@ targets:
       tags: >
         ["devicelab", "hostonly"]
       task_name: windows_startup_test
+
+  - name: Windows flutter_tool_startup
```
Comment (keyonghan) [inline .ci.yaml:4445]:
```suggestion
  - name: Windows flutter_tool_startup__windows
```
I should catch this earlier: the devicelab target name follows `<platform> <task_name>`.

File: .ci.yaml [hunk 3]
```diff
@@ -4441,3 +4441,33 @@ targets:
...
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      tags: >
+        ["devicelab", "hostonly"]
+      task_name: flutter_tool_startup__windows
+
+  - name: Linux flutter_tool_startup
```
Comment (keyonghan) [inline .ci.yaml:4455]: ditto

File: .ci.yaml [hunk 4]
```diff
@@ -4441,3 +4441,33 @@ targets:
...
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      tags: >
+        ["devicelab", "hostonly"]
+      task_name: flutter_tool_startup__linux
+
+  - name: Mac flutter_tool_startup
```
Comment (keyonghan) [inline .ci.yaml:4465]: ditto

Comment (jensjoha) [reply .ci.yaml]: Done.

File: .ci.yaml (hunk 3 above)
Comment (jensjoha) [reply .ci.yaml]: Done.

File: .ci.yaml (hunk 2 above)
Comment (jensjoha) [reply .ci.yaml]: Done.

**Review by keyonghan — APPROVED** [review]
LGTM

**Comment by jensjoha** [comment]
> > I don't know if there's any way of verifying that it works, though?
> 
> You can validate in different ways:
> 
> 1. locally: https://github.com/flutter/flutter/tree/master/dev/devicelab#running-tests-locally
> 2. via `led` run: go/flutter-luci-recipes#testing-a-local-change-for-a-new-builderrecipe

I have tested it locally (on Linux) --- but from there to knowing that it shows up correctly somewhere (with being tested on actual Windows, Linux and Mac etc) is quite a leap =)

File: .ci.yaml [hunk 5]
```diff
@@ -4441,3 +4441,33 @@ targets:
...
+
+  - name: Mac flutter_tool_startup__mac
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      tags: >
+        ["devicelab", "hostonly"]
+      task_name: flutter_tool_startup__mac
```
Comment (keyonghan) [inline .ci.yaml:4473]:
```suggestion
      task_name: flutter_tool_startup__macos
```
This needs to match the test file: `flutter_tool_startup__macos.dart`

Comment (jensjoha) [reply .ci.yaml]: Done. Thanks!

**Pull request merged**